### DISCUSSION TITLE
Corp Mining ledger parsing

### DISCUSSION THREAD

**Discussion by Raphyon** [Wilm0rien/omip#4] [Ideas]
Hi,

Love the work you have done so far to omip!
Would it be possible to add a way to automatically track the corp moon mining ledger?
What i mean is a page that pulls data from the corp mining ledger and automatically sums up the total mined ore per toon and or combines that with alts and thus can help calculate taxes if need be.

Considering we can already assign alts to toons in the corp you manage, i wouldnt think it would be a problem to connect the same logic to a mining ledger. And using the EVE ESI then pull market prices and calculate taxes as necessary.

**Comment by Wilm0rien** [comment]
hi thanks for the input!

I wanted to implement this already but when I looked through the ESI interfaces for mining it turned out to works like this:

1. the corp mining ledger is _only for corps who own their own moon_ and it will show you who took the ore from your moon. in this way the big alliances who own the moons can Tax every corp for respective ore mined. however _this is useless for 99% of the corps_ in the game because they do not own a moon. 
and this is why I did not implement the corp mining ledger. I would not even be able to test it if works. 

2. the individual user mining ledger shows the ores which have been mined with this particular character. because OMIP is only running on your local machine it is not possible to get the mining data from your corp members (you cannot access their ESI  tokens). The reason why this works with SEAT is that in contrast to OMIP all corp members enter their ESI keys on a _webserver_ .

because OMIP not a web-hosted application the only way I see to manage this is to implement a key-exchange mechanism: every corp member must install OMIP and generate an ESI access token which only allows reading the mining ledger. Since the only connection to exchange data via OMIP is the ESI server the keys are exchanged via encrypted ingame mail to the CEO / director of the corp. 

I was thinking about this but my corp members were not so happy about this idea and our mining operations are already tracked by the alliance SEAT server. that's why I did not implement it yet.

What do you think? are you a moon owner? do you only want the corp-mining ledger? would you pursuade your member to use OMIP to track their mining activity?

**Comment by Raphyon** [comment]
Hey hey!

So i was really just looking for a way to see what has been mined on the
specific moon.
I do own moons and dont have the brains myself to automate a process for
calculating moon tax.

Your software seemed perfect for my needs as i currently dont want to
implement ESI for my corp at this scale, but if you think this isn’t worth
the hassle then i completely understand that.

If it would be possible to just use the main/alt database already present
in the corp tab, to combine all the mined ores from that moon and pull the
market data. That would be amazing!

So in short, i just want the corp-only mining ledger and some kind of user
friendly logic to calculate the taxes.
😄

**Comment by Wilm0rien** [comment]
mhnmm, there are 14k+ moons in eve online and they should be owned by someone. 
I think it's ok to do something only for Moon-Owners!

I'll implement it within the next couple of weeks. but please note that I cannot test it with real in-game data!

**Comment by Wilm0rien** [comment]
one question you wrote:
>  sum up the total mined ore per toon

I looked into this and initially I wanted to create a table with character rows and ore columns but I found that there are 145 different ores in the game (for example four different uncommon moon ores with each commin in four flavours).

from the current tool design I have only space for 20 columns so I would like to aggregate this data.

I would suggest that you can generate a full table with all details via the copy CSV button and for an overview in the tool I calculate the value of the collected ores per month per main character by adding up the ISK value of all minerals/reactions contained in the total ore-quantity. the result would look like this table for moon ores https://ore.cerlestes.de/moon and this table for normal ores https://www.fuzzwork.co.uk/ore/

let me know if you have any doubts!

**Reply by Raphyon** [reply]
Hmmm, yes
I'm not sure how to display the data in the most simple way possible
But my intuition would like to think its possible to just sum up all the ores mined regardless of type or toon and just give a total assigned to the main account.

Sorry for not being constructive.
I can only see the end result in my head and not the background process :)

**Comment by Wilm0rien** [comment]
Hi Raphyon,

I have now a first working prototype:

1. extract the attached zip file [omip_V1.0.11.alpha.zip](https://github.com/Wilm0rien/omip/files/14472701/omip_V1.0.11.alpha.zip)
2. add your corporation director/CEO again to add esi-industry.read_corporation_mining.v1 to the allowed scopes
3. under keys tab select mining checkbox
4. run "update all data" from the file menu

* under your corporation tab you will see a mining tab if at least one of your own corp mates was mining on your moon
* in this tab you can choose between ISK ans ORE view
 * in ISK view you can define a percentage to calculate tax
 * in ORE view you see the total number of m3 mined per main per month
* if you click on a character it will show the details for this character and his alts per month

Please note: currently you can only see data about characters which are in the same corp as your director. I'm working on this now. 

please let me know if it works

**Reply by Raphyon** [reply]
Been testing it out now,

Seems the update never pulls mining data
I assume you are using the `[/corporation/{corporation_id}/mining/observers/{observer_id}/]` and
`[/corporation/{corporation_id}/mining/observers/]`
endpoints.

Maybe am i doing something wrong?

**Comment by Wilm0rien** [comment]
thanks for testing it and sorry for the late reply!

I have updated the tool to support external miners 
[omip_V1.0.11.alpha_V2.zip](https://github.com/Wilm0rien/omip/files/14550297/omip_V1.0.11.alpha_V2.zip)

currently the structure from which has been mined is not visible in the list this is an open point which I will work on next week.

regarding your problem with missing data:
it looks strongly like you did not check the mining checkbox in the key tab or you did not add the character again to unlock the esi-industry.read_corporation_mining.v1  scope

if you still don't get data please follow these steps:
1. update the tool (see link above)
2. open a commandline in the directory which contains omip.exe and open it with the -debug parameter
3. in the gui double check again the mining is checked in the keys tab
4. in your corporation tab there is a debug tab. click on the bottom icon "mining"
5. open an explorer (windows-key + e)  and put this in to the address bar "%appdata%\omip". hit enter to go to the omip directory 
6. open omip.log it should contain something like this

```
2024/03/10 14:09:29 REQ:
https://esi.evetech.net/v1/corporation/98179071/mining/observers?datasource=tranquility&page=1
2024/03/10 14:09:29 RESP: OK (2 bytes received)
2024/03/10 14:09:29 reading mining observers successful found 0 entries
```
in your case it should find more than 0 entries.

if it is still not there go to https://esi.evetech.net/ select Industry --> mining observers, authorize your chacter with  with esi-industry.read_corporation_mining.v1  and check if you get  a list with observers with the try button
try this again with the [/corporation/{corporation_id}/mining/observers/{observer_id}/] and enter an observer id which you have received from the first list. check if you see data there. 
let me know if you see the data in esi

**Comment by Wilm0rien** [comment]
ok there were several problems with the update lists which I simply forgot 

here is the latest fix:
[omip_V1.0.11.alpha_V3.zip](https://github.com/Wilm0rien/omip/files/14563915/omip_V1.0.11.alpha_V3.zip)

**Comment by Wilm0rien** [comment]
hi please check if this release works for you: https://github.com/Wilm0rien/omip/releases/tag/v1.0.11

so i can close this ticket

**Comment by Wilm0rien** [comment]
raphyon wrote 

> new release works perfectly 

this feature request is closed now
